Puppetfile

```diff
@@ -87,3 +68,29 @@ mod 'erwbgy/limits', '0.3.1'
 
 # For managing sysctl configuration
 mod 'herculesteam-augeasproviders_sysctl', '2.2.1'
+
+##### The following custom puppet modules must be specified with 1 attribute per line
+# Example:
+#  mod 'modulename'
+#    :git => <git URL>,
+#    :ref => <git ref>,
+#####
+
+# Dependencies for the Puppet IRC report processor, using our forked version
+# which updates on any changed status
+mod 'irc',
+  :git => 'https://github.com/jenkins-infra/puppet-irc.git',
+  :ref => '4e5e437'
+# Needed for managing our accounts in hiera, this fork contains the pull
+# request which adds support for multiple SSH keys:
+# <https://github.com/torrancew/puppet-account/pull/18>
+mod 'account',
+  :git => 'https://github.com/jenkins-infra/puppet-account.git',
+  :ref => '1deebe9'
+mod 'jenkins_keys',
+  :git => 'git@github.com:jenkins-infra/jenkins-keys.git',
```

Comment:
It's a constraint with the Git authentication: the puppetmaster uses SSH authentication with a "deploy key" defined on the repository jenkins-infra/jenkins-keys.

Rationale was because an Oauth token would have add too much permissions (no filtering by repository) while an SSH key is defined at a repository-level, with either read-only or read/write on the content (and no GitHub API access).

I'm not sure if using a GitHub app would be possible today: we would need to implement a hook in Puppetmaster's lifecycle, hence keeping the SSH.


Beside, it is a private repository that is removed from the `.fixtures.yml` so ci.jenkins.io never clones it.